realm/src/main/java/io/realm/Realm.java

```diff
@@ -396,6 +397,36 @@ public static Realm getInstance(Context context, String fileName, byte[] key) {
     }
 
     /**
+     * Realm static constructor that returns an un-persisted in-memory Realm instance.
+     * The identifier used to create this instance can be used to access the same in-memory Realm from multiple threads.
+     * {@link #close()} must be called when you are done using the Realm instance.
+     * <p>
+     * It sets auto-refresh on if the current thread has a Looper, off otherwise.
+     * <p>
+     * Note that Because in-memory Realms are not persisted, you must be sure to hold on to a non-closed reference to the
+     * Realm object returned from this (or from {@link #getInstance(RealmConfiguration)} with
+     * {@link RealmConfiguration.Builder#inMemory()} and a same identifier) for as long as you want the data
+     * to last. If all in-memory Realms with the same identifier have been closed, the in-memory Realm database will be
+     * released.
+     *
+     * @param context an Android {@link android.content.Context}
+     * @param identifier A string used to identify a particular in-memory Realm.
+     * @return an instance of the Realm class
+     *
+     * @throws IllegalArgumentException      An realm instance with same identifier/filename and different
+     *                                       {@link io.realm.internal.SharedGroup.Durability} has been created already.
+     * @throws RealmMigrationNeededException The model classes have been changed and the Realm
+     *                                       must be migrated
+     * @throws RealmIOException              Error when accessing underlying file
+     * @throws RealmException                Other errors
+     */
+    public static Realm getInMemoryInstance(Context context, String identifier) {
```

Comment:
Delete this constructor. People should use `Realm.getInstance(RealmConfiguration)` instead.